File: NovaMartPOS.Domain/Entities/Customer.cs
namespace NovaMartPOS.Domain.Entities;

public class Customer : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public int LoyaltyPoints { get; set; }
    public bool IsActive { get; set; } = true;
}

File: nova-mart-frontend/src/types/customer.ts
export interface Customer {
    id: number;
    name: string;
    phone: string | null;
    email: string | null;
    loyaltyPoints: number;
    isActive: boolean;
  }
  
  export interface CreateCustomerRequest {
    name: string;
    phone?: string | null;
    email?: string | null;
  }
  
  export interface UpdateCustomerRequest extends CreateCustomerRequest {
    isActive: boolean;
  }